### PR TITLE
Fix bugs, refactor alot of stuff

### LINKED ISSUE #67: Refactor UI to make it consistent

### PR DESCRIPTION
**Pull request by yujinyang1998**

### CODE REVIEW

**Review by CFZeon — APPROVED** [review]
lgtm!

**Comment by tienkhoa16** [comment]
Great work in refactoring! 👍

File: src/main/java/commands/diet/dietmanager/DietSessionClear.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ public void execute(String input, DietStorage storage) {
 
     private boolean checkConfirmation() {
```
Comment (tienkhoa16) [inline src/main/java/commands/diet/dietmanager/DietSessionClear.java:42]: This Confirmation method is duplicated in multiple places, so maybe we can extract them out into utils?

File: src/main/java/ui/diet/dietmanager/DietManagerUi.java [hunk 2]
```diff
@@ -29,27 +31,28 @@ public static void printBye() {
...
-            Pattern pattern = Pattern.compile("\\d{4}-\\d{2}-\\d{2}");
-            Matcher matcher = pattern.matcher(dateString);
-            boolean isValidDate = matcher.find();
-            String match = matcher.group();
-            return LocalDate.parse(match).format(DateTimeFormatter.ofPattern("MM-dd-yyyy"));
+            //Pattern pattern = Pattern.compile("\\d{4}-\\d{2}-\\d{2}");
+            //Matcher matcher = pattern.matcher(dateString);
+            //String match = matcher.group();
+            return DateParser.parseDate(dateString).format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+            //return LocalDate.parse(match).format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
```
Comment (tienkhoa16) [inline src/main/java/ui/diet/dietmanager/DietManagerUi.java]: Should we delete those commented out lines?

File: src/main/java/ui/workout/workoutsession/WorkoutSessionUi.java [hunk 3]
```diff
@@ -25,7 +25,7 @@ public static void emptyListError() {
     }
 
     public static void deleteFormatError() {
-        printer.showToUser("Wrong format, please enter in the format: delete [move ID]");
+        printer.showToUser("Wrong format, please enter in the format:\n\t delete [move ID]");
```
Comment (tienkhoa16) [inline src/main/java/ui/workout/workoutsession/WorkoutSessionUi.java]: For consistency in printing, can change here to "delete [MOVE_ID]"

File: src/main/java/ui/workout/workoutsession/WorkoutSessionUi.java [hunk 4]
```diff
@@ -48,11 +48,11 @@ public static void printHelp() {
     }
 
     public static void emptyInputError() {
-        System.out.println("Please enter something.");
+        printer.showToUser("Please enter something.");
     }
 
     public static void searchInputError() {
-        printer.showToUser("Wrong format, please enter in the format: search [name of move]");
+        printer.showToUser("Wrong format, please enter in the format:\n\t search [name of move]");
```
Comment (tienkhoa16) [inline src/main/java/ui/workout/workoutsession/WorkoutSessionUi.java]: And also here to "search [NAME_OF_MOVE]"

**Review by tienkhoa16 — COMMENTED** [review]
LGTM! Some refactoring suggested below can be considered

File: src/main/java/commands/diet/dietmanager/DietSessionClear.java (hunk 1 above)
Comment (yujinyang1998) [reply src/main/java/commands/diet/dietmanager/DietSessionClear.java:42]: @zsk612 @CFZeon

File: src/main/java/ui/diet/dietmanager/DietManagerUi.java (hunk 2 above)
Comment (yujinyang1998) [reply src/main/java/ui/diet/dietmanager/DietManagerUi.java]: will do

File: src/main/java/ui/workout/workoutsession/WorkoutSessionUi.java (hunk 3 above)
Comment (yujinyang1998) [reply src/main/java/ui/workout/workoutsession/WorkoutSessionUi.java]: Good idea

File: src/main/java/ui/workout/workoutsession/WorkoutSessionUi.java (hunk 4 above)
Comment (yujinyang1998) [reply src/main/java/ui/workout/workoutsession/WorkoutSessionUi.java]: will do

File: src/main/java/commands/diet/dietmanager/DietSessionClear.java (hunk 1 above)
Comment (tienkhoa16) [reply src/main/java/commands/diet/dietmanager/DietSessionClear.java:42]: Actually, it's used in workout manager and workout session as well. So can merge this PR first then I'll create a util method for that and you guys refactor your code accordingly

**Review by tienkhoa16 — APPROVED** [review]

**Pull request merged**